applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -1770,14 +1771,8 @@ namespace Kratos
 		}
 
 		if (fabs(v1[2]) != 1.00) {
-
-			v2 = MathUtils<double>::CrossProduct(v1, GlobalZ);
-			VectorNorm = MathUtils<double>::Norm(v2);
-			if (VectorNorm > numerical_limit) v2 /= VectorNorm;
-
-			v3 = MathUtils<double>::CrossProduct(v2, v1);
-			VectorNorm = MathUtils<double>::Norm(v3);
-			if (VectorNorm > numerical_limit) v3 /= VectorNorm;
+                    MathUtils<double>::UnitCrossProduct<arraydim, arraydim, arraydim>(v2, GlobalZ, v1);

```

Comment:
just to tell that it would automatically detect the template parameters. you do not need to put them explicitly (it is correct if you do of course)

Reply:
https://stackoverflow.com/questions/6359592/c-auto-detection-of-template-arguments You cannot...